packages/comms/src/ecl/xsdParser.ts

```diff
@@ -88,31 +88,31 @@ export class XSDXMLNode extends XSDNode {
                 break;
             default:
                 const numStr: string = "0123456789";
-                const underbarPos: number = this.type.lastIndexOf("_");
-                const length: number = underbarPos > 0 ? underbarPos : this.type.length;
+                const underbarPos: number = this.type?.lastIndexOf("_") ?? 0;
+                const length: number = underbarPos > 0 ? underbarPos : this.type?.length ?? 0;
                 let i: number = length - 1;
                 for (; i >= 0; --i) {
-                    if (numStr.indexOf(this.type.charAt(i)) === -1)
+                    if (numStr.indexOf(this.type!.charAt(i)) === -1)
                         break;
                 }
                 if (i + 1 < length) {
-                    retVal = parseInt(this.type.substring(i + 1, length), 10);
+                    retVal = parseInt(this.type!.substring(i + 1, length), 10);
```

Comment:
these two !s seem like they're conflicting with the lines right above and below here where ? is getting used for this.type